app/bot/broadcast_status.go

```diff
@@ -0,0 +1,115 @@
+package bot
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+)
+
+const (
+	MsgBroadcastStarted  = "Вещание началось"
+	MsgBroadcastFinished = "Вещание завершилось"
+)
+
+// BroadcastStatus bot returns current broadcast status
+type BroadcastStatus struct {
+	// Params
+	broadcastUrl string        // Url for "ping"
+	pingInterval time.Duration // Ping interval
+	delayToOff   time.Duration // After this interval of not OK, status will be switcher to OFF
+
+	// State
+	status         bool  // current broadcast status
+	lastSentStatus *bool // by default this value is null to react on first message
+	offPeriod      time.Duration
+	lastCheck      time.Time
+
+	// ping func (for tests)
+	ping func(ctx context.Context, url string) bool
+}
+
+// NewBroadcastStatus starts status checking goroutine and returns bot instance
+func NewBroadcastStatus(ctx context.Context, broadcastUrl string, pingInterval time.Duration, delayToOff time.Duration) *BroadcastStatus {
+	log.Printf("[INFO] BroadcastStatus bot with %v", broadcastUrl)
+	b := &BroadcastStatus{broadcastUrl: broadcastUrl, pingInterval: pingInterval, delayToOff: delayToOff, lastCheck: time.Now(), ping: ping}
+	go b.checker(ctx)
+	return b
+}
+
+// OnMessage returns current broadcast status
+func (b *BroadcastStatus) OnMessage(_ Message) (response string, answer bool) {
+	if b.lastSentStatus != nil && b.status == *b.lastSentStatus {
+		return
+	}
+
+	answer = true
+	response = MsgBroadcastStarted
+
+	status := b.status
+	b.lastSentStatus = &status
```

Comment:
both b.lastSeenStatus and b.status accessed from different goroutines without any protection against races. I bet running tests with -race will scream about it. I'd suggest protecting the access with mutex.

Reply:
Fixed